### ISSUE TITLE
SC2094 triggers on double loops

### ISSUE THREAD

**Issue by oliv3r** [koalaman/shellcheck#1368]
#### For bugs
- Rule Id: SC2094
- My shellcheck version: online
- [x] I read the issue's wiki page, e.g. https://github.com/koalaman/shellcheck/wiki/SC2094
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

while read -r dummy; do
  while read -r dummy2; do
    echo "${dummy}" "${dummy2}"
  done < /tmp/file
done < /tmp/file
```

#### Here's what shellcheck currently says:
Line 6:
  done < /tmp/file
         ^-- SC2094: Make sure not to read and write the same file in the same pipeline.
 
Line 7:
done < /tmp/file
       ^-- SC2094: Make sure not to read and write the same file in the same pipeline.

#### Here's what I wanted or expected to see:
I think at least on the wiki this 'exception' should be mentioned as well, where I think it's perfectly fine to read a file multiple times without reading it. Sadly, shell check can't detect the difference between input or output, so I don't think this is easily fixed, but a note in the exceptions section on the wiki is probably not a bad idea.

**Comment by ChillerDragon** [comment]
Also has false positives on function calls

```bash
#!/bin/bash

file=foo.txt

log() {
        echo "$1"
}

while read -r line
do
        echo "$file" # throws nothing
        log "$file" # throws SC2094
done < "$file"
```
